build/templates/_library_interpreter.py/numpy_write_method.py.mako

```diff
@@ -0,0 +1,44 @@
+<%page args="f, config, method_template"/>\
+<%
+    '''Renders a Intermediate method for writing numpy.array corresponding to the passed-in function metadata.'''
+
+    import build.helper as helper
+
+    parameters = f['parameters']
+    param_names_method = helper.get_params_snippet(f, helper.ParameterUsageOptions.LIBRARY_INTERPRETER_METHOD_DECLARATION)
+    param_names_library = helper.get_params_snippet(f, helper.ParameterUsageOptions.LIBRARY_METHOD_CALL)
+    param_ctypes_library = helper.get_params_snippet(f, helper.ParameterUsageOptions.CTYPES_ARGTYPES)
+
+    ivi_dance_parameters = helper.filter_ivi_dance_parameters(f)
+    ivi_dance_size_parameter = helper.find_size_parameter(ivi_dance_parameters, parameters)
+    len_parameters = helper.filter_len_parameters(f)
+    len_size_parameter = helper.find_size_parameter(len_parameters, parameters)
+    assert ivi_dance_size_parameter is None or len_size_parameter is None, str(f)
```

Comment:
I'm guessing this was for debugging?

Reply:
(The str(f))

Reply:
Yep. But even worse, all this ivi_dance stuff is left over from a copy/paste from when I was doing all the numpy stuff without the use of fancy methods at the LibraryInterpreter layer. The numpy functions don't use ivi_dance. So I've removed all the ivi_dance code from them.